### PR TITLE
Feature useToast

### PR DESCRIPTION
**Pull request by guilhermedecampo**

### CODE REVIEW

File: packages/toast/useTimeout.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import * as React from 'react'
```
Comment (grsabreu) [inline packages/toast/useTimeout.ts]: I supposed `* as React, { useEffect, useRef }` doesn't work right? 🤔

File: packages/toast/useTimeout.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  const savedCallback = React.useRef(null)
+
+  React.useEffect(() => {
+    savedCallback.current = callback
+  }, [callback])
+
+  React.useEffect(() => {
+    const tick = () => savedCallback.current()
+
+    if (delay === null) return
```
Comment (grsabreu) [inline packages/toast/useTimeout.ts]: if no delay the intuitive thing is indeed never call the callback?

Comment (grsabreu) [reply packages/toast/useTimeout.ts]: why ref is needed?

File: packages/toast/useToast.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import * as React from 'react'
+import { MdCheckCircle, MdError, MdWarning } from 'react-icons/md'
+
+import { useToastContext } from './useToastContext'
+
+const ToastWrapper = ({ children, color }) => (
+  <div
+    style={{
```
Comment (grsabreu) [inline packages/toast/useToast.tsx]: 💅 ?

Comment (grsabreu) [reply packages/toast/useToast.tsx]: styled-components not working fine here?

**Comment by grsabreu** [comment]
how this looks like? (visually)

File: packages/toast/useTimeout.ts (hunk 1 above)
Comment (guilhermedecampo) [reply packages/toast/useTimeout.ts]: yep

File: packages/toast/useToast.tsx (hunk 3 above)
Comment (guilhermedecampo) [reply packages/toast/useToast.tsx]: not working indeed

File: packages/toast/useTimeout.ts (hunk 2 above)
Comment (guilhermedecampo) [reply packages/toast/useTimeout.ts]: to keep the ref between  ♺

**Pull request merged**